tests/Functional/CorporateFooterRenderTest.php

```diff
@@ -0,0 +1,435 @@
+    $items = $data['legal_navigation'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'core', $expected);
+    }
+
+    // Update settings, assert footer changed.
+    $this->updateSiteSettings();
+    $this->getSession()->reload();
+
+    $actual = $assert->elementExists('css', 'div.ecl-footer-core__description');
+    $expected = new FormattableMarkup('This site is managed by the @name', ['@name' => 'ACP–EU Joint Assembly']);
+    $this->assertEquals($expected, $actual->getText());
+  }
+
+  /**
+   * Test European Commission footer standardised block rendering.
+   */
+  public function testEcFooterStandardisedBlockRendering(): void {
+    $this->configFactory->getEditable('oe_theme.settings')->set('branding', 'standardised')->save();
+    $data = $this->getFixtureContent('ec_footer.yml');
+    $this->renderCorporateBlocksFooter('ec', $data);
+
+    $this->getSession()->reload();
+    $assert = $this->assertSession();
+
+    // Make sure that footer block is present.
+    $this->assertFooterPresence('standardised', 7);
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-standardised section.ecl-footer-standardised__section1');
+
+    $actual = $section->find('css', 'a.ecl-footer-standardised__title');
+    $this->assertEquals('Drupal', $actual->getText());
+    $this->assertEquals('http://web:8080/build/', $actual->getAttribute('href'));
+
+    // Site owner is not set yet, lets make sure we don't have a description.
+    $assert->elementNotExists('css', 'div.ecl-footer-standardised__description');
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-standardised section.ecl-footer-standardised__section7');
+    $actual = $section->find('css', 'a.ecl-footer-standardised__title');
+    $this->assertEquals($data['corporate_site_link']['label'], $actual->getText());
+    $this->assertEquals($data['corporate_site_link']['href'], $actual->getAttribute('href'));
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-standardised section.ecl-footer-standardised__section8');
+    $items = $data['service_navigation'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'standardised', $expected);
+    }
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-standardised section.ecl-footer-standardised__section9');
+    $items = $data['legal_navigation'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'standardised', $expected);
+    }
+
+    // Update settings, assert footer changed.
+    $this->updateSiteSettings();
```

Comment:
This is similar to the issue we had outlined in a previous review: the caller needs to pass to the method the values to set, even if it's used once, or else the caller needs to know what that method does.